--- drivers/media/dvb-core/dvbdev.c
+++ drivers/media/dvb-core/dvbdev.c
@@ -83,8 +83,11 @@ static int dvb_device_open(struct inode *inode, struct file *file)
 			file->f_op = old_fops;
 			goto fail;
 		}
-		if(file->f_op->open)
+		if(file->f_op->open) {
+			mutex_unlock(&dvbdev_mutex);
 			err = file->f_op->open(inode,file);
+			mutex_lock(&dvbdev_mutex);
+		}
 		if (err) {
 			fops_put(file->f_op);
 			file->f_op = fops_get(old_fops);
--- drivers/media/dvb-core/dvbdev.h	2013-09-02 22:46:10.000000000 +0200
+++ drivers/media/dvb-core/dvbdev.h	2014-05-13 09:31:14.000000000 +0200
@@ -30,8 +30,10 @@
 
 #define DVB_MAJOR 212
 
-#if defined(CONFIG_DVB_MAX_ADAPTERS) && CONFIG_DVB_MAX_ADAPTERS > 0
-  #define DVB_MAX_ADAPTERS CONFIG_DVB_MAX_ADAPTERS
+#if defined(FFCONFIG_DVB_MAX_ADAPTERS) && FFCONFIG_DVB_MAX_ADAPTERS > 0
+  #define DVB_MAX_ADAPTERS FFCONFIG_DVB_MAX_ADAPTERS
+#elif defined(CONFIG_DVB_MAX_ADAPTERS) && CONFIG_DVB_MAX_ADAPTERS > 0
+  #define DVB_MAX_ADAPTERS CONFIG_DVB_MAX_ADAPTERS
 #else
   #define DVB_MAX_ADAPTERS 8
 #endif
